src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -338,8 +340,56 @@ private bool CanInstallPackage()
                 return false;
             }
 
+            return true;
+        }
+
+        public string ErrorMessage
+        {
+            get { return _errorMessage; }
+            set { Set(ref _errorMessage, value); }
+        }
+
+        public bool IsLibraryInstallationStateValid()
+        {
+            LibraryInstallationState libraryInstallationState = new LibraryInstallationState
+            {
+                LibraryId = PackageId,
+                ProviderId = SelectedProvider.Id,
+                DestinationPath = InstallationFolder.DestinationFolder,
+                Files = SelectedFiles?.ToList()
+            };
+
+            if (libraryInstallationState == null)
+            {
+                ErrorMessage = PredefinedErrors.UnknownError().Message;
+                return false;
+            }
```

Comment:
I personally wouldn't bother, if the allocation fails, the system is dead.

Reply:
Hmmm, I didn't get you. Did you mean the null check is unnecessary?

---
In reply to: [201158730](https://github.com/aspnet/LibraryManager/pull/293#discussion_r201158730) [](ancestors = 201158730)

Reply:
It's no more necessary than checking if errors is null after the new List<IError> statement after it.

---
In reply to: [201159352](https://github.com/aspnet/LibraryManager/pull/293#discussion_r201159352) [](ancestors = 201159352,201158730)